client/extensions/relatedArticles/src/EventsRelatedArticles/EventsRelatedArticlesModal.tsx

```diff
@@ -0,0 +1,307 @@
+import React from 'react';
+import {
+    SearchBar,
+    Modal,
+    Dropdown,
+    Spacer,
+    Button,
+    WithPagination,
+    Loader,
+    Panel,
+    PanelHeader,
+    PanelContent,
+    PanelContentBlock,
+    ContentDivider,
+} from 'superdesk-ui-framework/react';
+import {getProjectedFieldsArticle, gettext} from 'superdesk-core/scripts/core/utils';
+import {httpRequestJsonLocal} from 'superdesk-core/scripts/core/helpers/network';
+import {toElasticQuery} from 'superdesk-core/scripts/core/query-formatting';
+import {IArticle, IRestApiResponse, ISuperdeskQuery} from 'superdesk-api';
+import {PreviewArticle} from './PreviewArticle';
+import {cleanArticlesFields} from './utils';
+import {RelatedArticlesListComponent} from './RelatedArticlesListComponent';
+
+interface IProps {
+    closeModal: () => void;
+    selectedArticles?: Array<Partial<IArticle>>;
+    onChange: (value: Array<Partial<IArticle>>) => void;
+}
+
+interface IState {
+    articles: Array<Partial<IArticle>>;
+    searchQuery: string;
+    loading: boolean;
+    currentlySelectedArticles?: Array<Partial<IArticle>>;
+    activeLanguage: {code: string; label: string;};
+    previewItem: Partial<IArticle> | null;
+    repo: string | null;
+}
+
+
+export default class EventsRelatedArticlesModal extends React.Component<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            articles: [],
+            searchQuery: '',
+            loading: true,
+            currentlySelectedArticles: this.props.selectedArticles,
+            activeLanguage: {label: 'All languages', code: ''},
+            previewItem: null,
+            repo: null,
+        };
+    }
+
+    componentDidMount() {
+        httpRequestJsonLocal<IRestApiResponse<any>>({
+            method: "GET",
+            path: "/search_providers",
+            urlParams: {
+                manage: 1,
+            }
+        }).then((result) => {
+            const repoId = result._items.find((provider) => provider.source === 'http://wss-01.staging.belga.be:9000/archivenewsobjects')._id;
+
+            this.setState({
+                repo: repoId,
+            })
+        })
+    }
+
+    componentDidUpdate(_prevProps: Readonly<IProps>, prevState: Readonly<IState>): void {
+        if (prevState.activeLanguage.code !== this.state.activeLanguage.code
+            || prevState.searchQuery !== this.state.searchQuery
+            || prevState.repo !== this.state.repo
+        ) {
+            this.setState({
+                loading: true,
+            })
+        }
+    }
+
+    render(): React.ReactNode {
+        const {closeModal} = this.props;
+        const allLanguages = [
```

Comment:
This should ideally come from the languages defined in the system, see [vocabs](https://github.com/superdesk/superdesk-planning/blob/develop/client/selectors/vocabs.ts#L18) selector for an example